### ISSUE TITLE
buildDotnetPackage: xbuild tool is deprecated and will be removed in future updates, use msbuild instead

### ISSUE THREAD

**Issue by milahu** [NixOS/nixpkgs#162387]
`buildDotnetPackage` is using deprecated `xcode` build tool, should use `msbuild` tool

### Reproduce

build [subtitleedit](https://github.com/SubtitleEdit/subtitleedit) with

```nix
/*
nix-build -E 'with import <nixpkgs> { }; callPackage ./default.nix { }'
*/

{ lib
, fetchFromGitHub
, buildDotnetPackage
}:

buildDotnetPackage rec {
  pname = "subtitleedit";
  baseName = pname; # legacy
  version = "3.6.4";
  src = fetchFromGitHub {
    owner = "SubtitleEdit";
    repo = "subtitleedit";
    rev = version;
    sha256 = "5EW9qPSVbKRpWy2oR1zailECStWRYnzzGO6rxIRctN0=";
  };
  projectFile = "SubtitleEdit.sln";
}
```

> `building`
> `Building dotNET packages...`
> 
> `>>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<`
>
> ...
>
> `The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.`

"xbuild tool is deprecated" is just a warning

the build crashes at the "XML namespace" error,
which is caused by the source file [subtitleedit/src/libse/LibSE.csproj](https://github.com/SubtitleEdit/subtitleedit/blob/master/src/libse/LibSE.csproj).
this error disappears when i use `msbuild`

### Workaround

in buildPhase, replace xbuild with msbuild

```nix
/*
nix-build -E 'with import <nixpkgs> { }; callPackage ./default.nix { }'
*/

{ lib
, fetchFromGitHub
, buildDotnetPackage
, msbuild
}:

buildDotnetPackage rec {
  pname = "subtitleedit";
  baseName = pname; # legacy
  version = "3.6.4";
  src = fetchFromGitHub {
    owner = "SubtitleEdit";
    repo = "subtitleedit";
    rev = version;
    sha256 = "5EW9qPSVbKRpWy2oR1zailECStWRYnzzGO6rxIRctN0=";
  };
  projectFile = "SubtitleEdit.sln";
  buildInputs = [
    msbuild
  ];
  buildPhase = ''
    msbuild /p:Configuration=Release ${projectFile}
  '';
}
```

### Nixpkgs files

buildDotnetPackage = [nixpkgs/pkgs/build-support/dotnet/build-dotnet-package/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/dotnet/build-dotnet-package/default.nix)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.99, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0pre20220127_558c4ee`
 - channels(root): `"nixos-21.11pre331460.931ab058daa"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by milahu** [comment]
ping @IvarWithoutBones

**Comment by IvarWithoutBones** [comment]
Yeah, `buildDotnetPackage` is pretty ancient. I've been planning to rewrite it for a while now, or potentially combining it with `buildDotnetModule` since I think we can reuse a few things. I'll get on that after all major `buildDotnetModule` functionality is implemented :+1:

**Comment by milahu** [comment]
for the record, i gave up at

```
NETSDK1064: Package UTF.Unknown, version 2.5.0 was not found
```

i add the library with

```sh
        ln -s -v ${UTF.Unknown}/lib/dotnet/UTF.Unknown $d/packages/UTF.Unknown.${UTF.Unknown.version}
```

but its not found by msbuild.
i tried to debug with `msbuild -verbosity:diagnostic` and `strace -ff msbuild`, but no help

https://github.com/milahu/nur-packages/tree/master/pkgs/subtitleedit
